test/TestDriver/index.ts

```diff
@@ -384,93 +197,278 @@ export abstract class RavenTestDriver implements IDisposable {
         return Promise.resolve(result);
     }
 
-    public waitForUserToContinueTheTest(store: DocumentStore): void {
-        // TODO
+    public async waitForValue<T>(act: () => Promise<T>, expectedValue: T, opts: { timeout?: number; equal?: (a: T, b: T) => boolean } = {}) {
+        const timeout = opts.timeout ?? 15_000;
+        const identity = (a, b) => a === b;
+        const compare = opts.equal ?? identity;
+
+        const sw = Stopwatch.createStarted();
+
+        do {
+            try {
+                const currentVal = await act();
+                if (compare(expectedValue, currentVal)) {
+                    return currentVal;
+                }
+
+                if (sw.elapsed > timeout) {
+                    return currentVal;
+                }
+            } catch (e) {
+                if (sw.elapsed > timeout) {
+                    throwError("InvalidOperationException", e);
+                }
+            }
+            await delay(16);
+        } while (true);
+    }
+
+    protected static _killProcess(p: ChildProcess) {
+        if (p && !p.killed) {
+            log.info("Kill global server");
 
-        // String databaseNameEncoded = UrlUtils.escapeDataString(store.getDatabase());
-        // String documentsPage = 
-        //      store.getUrls()[0] + "/studio/index.html#databases/documents?&database=" 
-        //      + databaseNameEncoded + "&withStop=true";
+            try {
+                p.kill();
+            } catch (err) {
+                log.error(err);
+            }
+        }
+    }
 
-        // openBrowser(documentsPage);
+    public async waitForUserToContinueTheTest(store: IDocumentStore) {
+        const databaseNameEncoded = encodeURIComponent(store.database);
+        const documentsPage = store.urls[0] + "/studio/index.html#databases/documents?&database="
+            + databaseNameEncoded + "&withStop=true";
 
-        // do {
-        //     try {
-        //         Thread.sleep(500);
-        //     } catch (InterruptedException ignored) {
-        //     }
+        this._openBrowser(documentsPage);
 
-        //     try (IDocumentSession session = store.openSession()) {
-        //         if (session.load(ObjectNode.class, "Debug/Done") != null) {
-        //             break;
-        //         }
-        //     }
+        do {
+            await delay(500);
 
-        // } while (true);
+            const session = store.openSession();
+            if (await session.load("Debug/Done")) {
+                break;
+            }
+        } while (true);
     }
 
     protected _openBrowser(url: string): void {
         // tslint:disable-next-line:no-console
         console.log(url);
 
         if (os.platform() === "win32") {
-            spawn("powershell.exe", ["-c", `start-process ${url}`], {
-                detached: true
-            });
+            // noinspection JSIgnoredPromiseFromCall
+            open(url);

```

Comment:
What's wrong with Powershell?

Reply:
it doesn't work properly - using `open` lib which works out of box.

Reply:
why? what's the error